server/routes/server.routes.js

```diff
@@ -20,11 +21,17 @@ router.get('/signup', (req, res) => {
 });
 
 router.get('/projects/:project_id', (req, res) => {
-  res.send(renderIndex());
+  projectExists(req.params.project_id, exists => (
+    exists ? res.send(renderIndex()) : get404Sketch(html => res.send(html))
+  ));
 });
 
 router.get('/:username/sketches/:project_id', (req, res) => {
-  res.send(renderIndex());
+  userExists(req.params.username, exists => (
```

Comment:
can you also check if the sketch belongs to the user here?

Reply:
done :) Tell me if there are any other mistakes!